package.json

```diff
@@ -1525,62 +1525,62 @@
                 {
                     "command": "aws.s3.uploadFileToParent",
-                    "when": "view == aws.explorer && viewItem == awsS3FileNode",
+                    "when": "view == aws.explorer && viewItem =~ /^awsS3FileNode[-ANU]*$/",
                     "group": "1@1"
                 },
                 {
                     "command": "aws.s3.createFolder",
-                    "when": "view == aws.explorer && viewItem =~ /^(awsS3BucketNode|awsS3FolderNode)$/",
+                    "when": "view == aws.explorer && viewItem =~ /^(awsS3BucketNode|awsS3FolderNode)[-ANU]*$/",
                     "group": "1@1"
                 },
                 {
                     "command": "aws.iot.deactivateCert",
-                    "when": "view == aws.explorer && viewItem =~ /^awsIotCertificateNode.(Things|Policies).ACTIVE$/",
+                    "when": "view == aws.explorer && viewItem =~ /^awsIotCertificateNode[-ANU]*.(Things|Policies).ACTIVE$/",
                     "group": "1@1"
                 },
                 {
                     "command": "aws.iot.activateCert",
-                    "when": "view == aws.explorer && viewItem =~ /^awsIotCertificateNode.(Things|Policies).INACTIVE$/",
+                    "when": "view == aws.explorer && viewItem =~ /^awsIotCertificateNode[-ANU]*.(Things|Policies).INACTIVE$/",
                     "group": "1@1"
                 },
                 {
                     "command": "aws.iot.revokeCert",
-                    "when": "view == aws.explorer && viewItem =~ /^awsIotCertificateNode.(Things|Policies).(ACTIVE|INACTIVE)$/",
+                    "when": "view == aws.explorer && viewItem =~ /^awsIotCertificateNode[-ANU]*.(Things|Policies).(ACTIVE|INACTIVE)$/",
                     "group": "1@2"
                 },
                 {
                     "command": "aws.iot.setDefaultPolicy",
-                    "when": "view == aws.explorer && viewItem == awsIotPolicyVersionNode.NONDEFAULT",
+                    "when": "view == aws.explorer && viewItem =~ /^awsIotPolicyVersionNode[-ANU]*.NONDEFAULT$/",
                     "group": "1@1"
                 },
                 {
                     "command": "aws.iot.copyEndpoint",
-                    "when": "view == aws.explorer && viewItem == awsIotNode",
+                    "when": "view == aws.explorer && viewItem =~ /^awsIotNode[-ANU]*$/",
                     "group": "2@1"
                 },
                 {
                     "command": "aws.copyName",
-                    "when": "view == aws.explorer && viewItem =~ /^(awsRegionFunctionNode|awsRegionFunctionNodeDownloadable|awsCloudFormationFunctionNode|awsStateMachineNode|awsCloudFormationNode|awsS3BucketNode|awsS3FolderNode|awsS3FileNode|awsApiGatewayNode|awsIotThingNode)$|^(awsAppRunnerServiceNode|awsIotCertificateNode|awsIotPolicyNode|awsIotPolicyVersionNode)/",
+                    "when": "view == aws.explorer && viewItem =~ /^aws[^-]*-([AU]*N)/",
```

Comment:
copyName is enabled by the viewItem name instead of enumerating every viewItem here.

Could also do this for "DISABLED"/"ENABLED".

Reply:
Another idea: use the same pattern as CSS classes and append dot-separated (or whatever other delimiter) keys to the end of a node's context value. Then you can pattern match for `/\.${contextKey}/`, e.g. `awsRegionFunctionNode.copyName.copyArn`

The resulting strings could get pretty big though.

Reply:
context value would be the natural mechanism, but AFAIR it doesn't work in cloud9.